soc/xtensa/esp32s2/linker.ld

```diff
@@ -0,0 +1,263 @@
+/*
```

Comment:
This file is identical i layout to this file:
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/xtensa/esp32/linker.ld

But with different values on some lines.

Could you align, so that a single file can be used ?
Similar to what is done for:
ARC: https://github.com/zephyrproject-rtos/zephyr/blob/master/include/arch/arc/v2/linker.ld
ARM: https://github.com/zephyrproject-rtos/zephyr/blob/master/include/arch/arm/aarch32/cortex_m/scripts/linker.ld
and others.

and then include, for example like here:
https://github.com/zephyrproject-rtos/zephyr/blob/8d563d65074c26e6a19cace753fe9558fd28f323/soc/arc/snps_arc_hsdk/linker.ld#L23
and
https://github.com/zephyrproject-rtos/zephyr/blob/8d563d65074c26e6a19cace753fe9558fd28f323/soc/arm/atmel_sam/sam3x/linker.ld#L9

Reply:
So, all Xtensa targets right now do have very similar linking scripts (by nature of being the same architecture), not just esp32 and esp32s2, and they do the same thing (for example, setup the vector table) in slightly different ways. I can look into unifying them in the same way ARC and ARM are; but it will require a bit more research into the possible configuration options since Xtensa is a pretty dynamic architecture dependent on the configuration settings when generating the IP.

Reply:
Should that be a separate PR on just the existing Xtensa architectures Zephyr supports or should that be part of this one?

Reply:
For aligning esp32 and esp32s2 then doing so in this PR fine.
A more general cleanup should be in independent cleanup.